sql/timeseries/a11yScores.sql

```diff
@@ -1,24 +1,25 @@
 #standardSQL
-CREATE TEMPORARY FUNCTION getA11yScore(report STRING)
+# Lighthouse changed format of scores in v3.0.0 released in July 2018 so hande old and new
+CREATE TEMPORARY FUNCTION getA11yScore(reportCategories STRING, score STRING)
 RETURNS FLOAT64
 LANGUAGE js AS """
-  $=JSON.parse(report);
-  if ($.reportCategories) {
-    return $.reportCategories.find(i => i.name === 'Accessibility').score;
+  if (reportCategories) {
+    $=JSON.parse(reportCategories);
+    return $.find(i => i.name === 'Accessibility').score;
   } else {
-    return $.categories.accessibility.score * 100;
+    return score * 100;
   }
 """;
 
 SELECT
   SUBSTR(_TABLE_SUFFIX, 0, 10) AS date,
   UNIX_DATE(CAST(REPLACE(SUBSTR(_TABLE_SUFFIX, 0, 10), '_', '-') AS DATE)) * 1000 * 60 * 60 * 24 AS timestamp,
   IF(ENDS_WITH(_TABLE_SUFFIX, 'desktop'), 'desktop', 'mobile') AS client,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(11)], 2) AS p10,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(26)], 2) AS p25,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(51)], 2) AS p50,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(76)], 2) AS p75,
-  ROUND(APPROX_QUANTILES(getA11yScore(report), 101)[OFFSET(91)], 2) AS p90
+  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 101)[OFFSET(11)], 2) AS p10,
+  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 101)[OFFSET(26)], 2) AS p25,
+  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 101)[OFFSET(51)], 2) AS p50,
+  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 101)[OFFSET(76)], 2) AS p75,
+  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 101)[OFFSET(91)], 2) AS p90
```

Comment:
For more precision we should use 1000 quantiles with percentile*10 as the offset. [APPROX_QUANTILES](https://cloud.google.com/bigquery/docs/reference/standard-sql/approximate_aggregate_functions#approx_quantiles) also returns n+1 elements, so no need to make it 101 or 1001.
```suggestion
  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 1000)[OFFSET(100)], 2) AS p10,
  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 1000)[OFFSET(250)], 2) AS p25,
  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 1000)[OFFSET(500)], 2) AS p50,
  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 1000)[OFFSET(750)], 2) AS p75,
  ROUND(APPROX_QUANTILES(getA11yScore(JSON_EXTRACT(report, '$.reportCategories'), JSON_EXTRACT(report, '$.categories.accessibility.score')), 1000)[OFFSET(900)], 2) AS p90
```

Reply:
Hey that wasn't my code! That was in there already. 😁

But fine - good to fix up while we're here.

Reply:
Actually does using 1000 matter in this case since there are only 100 distinct Lighthouse scores?

Still guess to follow a standard for others copying and pasting the code without realising this nuance in this particular case.

Reply:
Actually looks like in the old results, scores were more than two decimal places so will keep the 1000.